### DISCUSSION TITLE
Add alias for query string parameters

### DISCUSSION THREAD

**Discussion by jdrieghe** [livewire/livewire#2496] [3. Feature ideas]
It's currently possible to expose component properties as query string parameters and add the additional option `except` to avoid adding a parameter like `page=1` when that is the default. 
When migrating an older Angular application to livewire, I ran into a hyphenated query string parameter. To ensure backwards compatibility, I had to introduce some URL rewriting.
Would it be interesting to add an `alias` option (or something similar, e.g. `as`) to the `$queryString` array so that the component's internal property name can be different from what is exposed in the query string?

I would be happy to contribute a PR for this and add the necessary documentation.

**Comment by HelgeSverre** [comment]
Had the same idea, I would love for something like this to be possible, it would help keep the the URL short and sweet, while still being able to refer to properties with sensible names:

![image](https://user-images.githubusercontent.com/1089652/118258624-695edb00-b4b0-11eb-8215-820b5db9c5c2.png)

**Reply by jcc5018** [reply]
Any progress on this functionality? Also, perhaps being able to use a slug instead of id

**Comment by joshhanley** [comment]
A PR #4178 for this has been submitted and merged!
